codec-http2/src/test/java/io/netty/handler/codec/http2/Http2FrameCodecTest.java

```diff
@@ -516,9 +529,23 @@ public void streamZeroWindowUpdateIncrementsConnectionWindow() throws Http2Excep
         int initialWindowSizeBefore = localFlow.initialWindowSize();
         Http2Stream connectionStream = connection.connectionStream();
         int connectionWindowSizeBefore = localFlow.windowSize(connectionStream);
-        // We only replenish the flow control window after the amount consumed drops below the following threshold.
-        // We make the threshold very "high" so that window updates will be sent when the delta is relatively small.
-        ((DefaultHttp2LocalFlowController) localFlow).windowUpdateRatio(connectionStream, .999f);
+
+        AtomicReference<Http2Exception> errorRef = new AtomicReference<>();
+        channel.eventLoop().execute(() -> {
```

Comment:
we need to call this on the `EventLoop` as the flowcontroller asserts that we are on the event loop.